recipe/recipe.yaml

```
@@ -0,0 +1,228 @@
+
+schema_version: 1
+
+context:
+  version: 2.3.1
+  # -1 means "not a dev version", a number > -1 will emit postfix "rc${{ rc_num }}"
+  rc_num: "-1"
+  # numpy will by default use the ABI feature level for the first numpy version
+  # that added support for the oldest currently-supported CPython version; see
+  # https://github.com/numpy/numpy/blob/v2.0.0rc1/numpy/_core/include/numpy/numpyconfig.h#L124
+  default_abi_level: 1.23
+  # ensure is_freethreading gets detected as a used variable
+  touch_is_freethreading: ${{ is_freethreading }}
+
+package:
+  name: numpy
+  version: ${{ version }}${{ "rc" + rc_num if rc_num != "-1" else "" }}
+
+source:
+  # The sdist distributed by numpy contains submodules; by taking the tarball
+  # from github we can everything pythran ourselves, but manually need to include
+  # the submodules (not in tarball due to dear-github/dear-github#214); for the
+  # list of modules see https://github.com/numpy/numpy/blob/main/.gitmodules
+  - url: https://github.com/numpy/numpy/archive/refs/tags/v${{ version }}.tar.gz
+    sha256: d040da6eb210b6e04cb76fa40b6ab221858ed9e188828ee8d8daa455cf41456f
+  # https://github.com/numpy/numpy/tree/v{{ version }}/numpy/_core/src
+  - target_directory: numpy/_core/src/highway
+    git: https://github.com/google/highway.git
+    rev: 0b696633f9ad89497dd5532b55eaa01625ad71ca
+  # https://github.com/numpy/numpy/tree/v{{ version }}/numpy/_core/src/common
+  - target_directory: numpy/_core/src/common/pythoncapi-compat
+    git: https://github.com/python/pythoncapi-compat.git
+    rev: 0f1d42a10a3f594ad48894912396df31b2c2d55d
+  # https://github.com/numpy/numpy/tree/v{{ version }}/numpy/_core/src/npysort
+  - target_directory: numpy/_core/src/npysort/x86-simd-sort
+    git: https://github.com/intel/x86-simd-sort.git
+    rev: c306ac581a59f89585d778254c4ed7197e64ba2d
+  # https://github.com/numpy/numpy/tree/v{{ version }}/numpy/_core/src/umath
+  - target_directory: numpy/_core/src/umath/svml
+    git: https://github.com/numpy/SVML.git
+    rev: 32bf2a98420762a63ab418aaa0a7d6e17eb9627a
+  # https://github.com/numpy/numpy/tree/v{{ version }}/numpy/fft
+  - target_directory: numpy/fft/pocketfft
+    git: https://github.com/mreineck/pocketfft.git
+    rev: 33ae5dc94c9cdc7f1c78346504a85de87cadaa12
+  # https://github.com/numpy/numpy/tree/v{{ version }}/vendored-meson
+  - target_directory: vendored-meson/meson
+    git: https://github.com/numpy/meson.git
+    rev: f754c4258805056ed7be09830d96af45215d341b
+
+build:
+  number: 1
+  skip: match(python, "<3.11")
+  python:
+    entry_points:
+      - if: win
+        then:
+          - f2py = numpy.f2py.f2py2e:main
+          - numpy-config = numpy._configtool:main
+
+requirements:
+  build:
+    - if: build_platform != target_platform
+      then:
+        - python
+        - cross-python_${{ target_platform }}
+        - cython
+        - ninja
+        - pkg-config
+    - ${{ compiler('c') }}
+    - ${{ stdlib('c') }}
```

Let's use this opportunity to order the stdlib before the compilers
```suggestion
    - ${{ stdlib('c') }}
    - ${{ compiler('c') }}
```

---

done